presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/InMemoryCachingHiveMetastore.java

```diff
@@ -730,7 +732,7 @@ private Map<KeyAndContext<HivePartitionName>, Optional<Partition>> loadPartition
         }
 
         ImmutableMap.Builder<KeyAndContext<HivePartitionName>, Optional<Partition>> partitions = ImmutableMap.builder();
-        Map<String, Optional<Partition>> partitionsByNames = delegate.getPartitionsByNames(firstPartitionKey.getContext(), databaseName, tableName, partitionsToFetch);
+        Map<String, Optional<Partition>> partitionsByNames = firstPartitionKey.context.getRuntimeStats().profileNanos(GET_HIVE_PARTITIONS_BY_NAMES_TIME_NANOS, () -> delegate.getPartitionsByNames(firstPartitionKey.getContext(), databaseName, tableName, partitionsToFetch));
```

Comment:
Instead of here... can you add it here ?
https://github.com/prestodb/presto/blob/master/presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java#L762